src/txn/txn.c

```diff
@@ -441,20 +441,58 @@ __wt_txn_config(WT_SESSION_IMPL *session, const char *cfg[])
 	if (cval.len > 0) {
 #ifdef HAVE_TIMESTAMPS
 		wt_timestamp_t ts;
+		WT_TXN_GLOBAL *txn_global;
+		char timestamp_buf[2 * WT_TIMESTAMP_SIZE + 1];
+		bool round_to_oldest;
 
+		txn_global = &S2C(session)->txn_global;
 		WT_RET(__wt_txn_parse_timestamp(session, "read", &ts, &cval));
-		WT_RET(__wt_timestamp_validate(session,
-		    &ts, &cval, true, false, false));
-		__wt_timestamp_set(&txn->read_timestamp, &ts);
+		WT_RET(__wt_config_gets_def(session,
+		    cfg, "round_to_oldest", 0, &cval));
+		round_to_oldest = false;
```

Comment:
This is a weird semantic - we pull the configuration value into cval, then assign an unconditional value to `round_to_oldest`. I can see that it's done so that `round_to_oldest` is always initialized for the verbose block below, but the code would be clearer if this initialization was done above in a local variable initialization block (like `txn_global`). Then the `__wt_config_gets_def` call deserves a comment explaining that we are parsing the configuration value now to avoid needing to do it while holding the `txn_global` lock.
It's a bit sad that all WT_TRANSACTION::begin calls will now have to take this lock and do some timestamp comparison - does this change introduce performance regressions? The check is really only for enforcing our API contract - maybe that could happen only in diagnostic mode.

Reply:
I talked to @bvpvamsikrishna directly about this. My performance concerns were unfounded - this change is replacing a call to validate timestamps that did the same locking.
My remaining concern is around the usage of `round_to_oldest` - and we discussed a resolution